tldw_Server_API/app/core/AuthNZ/pg_migrations_extra.py

```
@@ -2829,6 +3045,65 @@ async def ensure_org_provider_secrets_pg(pool: DatabasePool | None = None) -> bo
+            logger.debug(f"ensure_identity_federation_tables_pg: core table ensure skipped/failed: {exc}")
+
+        for ddl_group in (
+            _CREATE_IDENTITY_PROVIDERS,
+            _CREATE_FEDERATED_IDENTITIES,
+            _CREATE_FEDERATED_MANAGED_GRANTS,
+        ):
+            for sql, params in ddl_group:
+                try:
+                    await db_pool.execute(sql, *params)
+                except _PG_MIGRATIONS_NONCRITICAL_EXCEPTIONS as exc:
+                    logger.debug(f"PG ensure identity federation DDL failed: {exc}")
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)